src/serial.rs

```diff
@@ -0,0 +1,474 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Portions Copyright 2020 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+//
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Provides emulation for Linux serial console.
+//!
+//! This is done by emulating an UART 8250 serial port.
+
+use std::collections::VecDeque;
+use std::io::{Result, Write};
+
+use vmm_sys_util::eventfd::EventFd;
+
+// Register offsets.
+const DATA: u8 = 0;
+const IER: u8 = 1;
+const IIR: u8 = 2;
+const LCR: u8 = 3;
+const MCR: u8 = 4;
+const LSR: u8 = 5;
+const MSR: u8 = 6;
+const SCR: u8 = 7;
+
+const DLAB_LOW: u8 = 0;
+const DLAB_HIGH: u8 = 1;
+
+const LOOP_SIZE: usize = 0x40;
+
+const IER_RDA_BIT: u8 = 0x01;
+const IER_THR_BIT: u8 = 0x02;
+const IER_8250_BITS: u8 = 0x0F;
```

Comment:
What is the distinction between _BIT and _BITS? How many bits does _BITS refer to? :)) I that number that you defined here the beginning of the bit?

Reply:
Synced offline.